Added button.jsx file inside templates file

JohnAKASquib opened the pull request:
<!-- Please feel free to remove whatever sections/lines in this aren’t relevant.

Use the title line as the title of your pull request, then delete these lines.

## Title line template: [Title]: Brief description

Website: For pull requests that impact designsystem.digital.gov’s look, feel, or functionality, please open a pull request on the uswds-site repo (https://github.com/uswds/uswds-site).

-->

## Description

This pull request adds a button template file inside a folder titled __jsx-templates__. Button.jsx is a function that returns a string because Lamarr and I were getting the message: `Unexpected token '<'` When we tried to return the button using the format `return(<Button whatever=${config.whatever} etc... />)`. If we can stop that error from occuring we'll likely switch it back to this way. We've tested creating the button as it is now by passing a **config** object into the function and then passing the string returned into the **generator()** function as per usual.

JohnAKASquib committed:
added button.jsx file inside templates file

JohnAKASquib committed:
removed lines from previous commit that werent needed

JohnAKASquib commented on jsx-templates/button.jsx:
```diff
@@ -0,0 +1,29 @@
+    function focus(e) {
+      e.target.style.background = "red";
+    }
+    
```
get rid of these functions

JohnAKASquib commented on jsx-templates/button.jsx:
```diff
@@ -0,0 +1,29 @@
+      e.target.style.background = "red";
+    }
+    
+      <Button
```
should be HTML5 `<button>`

JohnAKASquib commented on jsx-templates/button.jsx:
```diff
@@ -0,0 +1,29 @@
+        onMouseEnter = {${config.onMouseEnter ? config.onMouseEnter : false}}
+        onMouseLeave = {${config.onMouseLeave ? config.onMouseLeave : false}}
+        onFocus = {${config.onFocus ? config.onFocus : false}}
+        inverse = {${config.inverse ? config.inverse : false}}
```
Anything that has to do with event handling just forget about

JohnAKASquib commented on jsx-templates/button.jsx:
```diff
@@ -0,0 +1,29 @@
+        outline = {${config.outline ? config.outline : false}}
+        unstyled = {${config.unstyled ? config.unstyled : false}}
+  
+      />`;
```
Again, HTML5 so it should be `<button>Name</button>`

JohnAKASquib commented on jsx-templates/button.jsx:
```diff
@@ -0,0 +1,29 @@
+
+exports.buttonFunc = function (config) {
+  return `
+    //functions
```
Do return( ) not return ``

JohnAKASquib commented on jsx-templates/button.jsx:
```diff
@@ -0,0 +1,29 @@
+//this template will live on USWDS repo
+//--sandbox for styles.css--//
+//https://codesandbox.io/s/bold-voice-xwend?file=/src/App.js
+
```
In the comments, put info about what gets passed into `<button>` and where it comes from

JohnAKASquib committed:
Changed attributes, added comments
No longer returning a template literal, now returning a usual
HTML5 button.
Common attributes all buttons need were added, and attributes and
functions that have to do with event handling were removed.
Added comments that are mostly from the MDN documentation for each
attribute (except for class).

Lamarr20 committed:
move jsx button template into src/components/button folder

Lamarr20 committed:
simplify attribute description

Lamarr20 committed:
add config usage

Lamarr20 committed:
add multiple class usage

Lamarr20 committed:
update documentation

mkcode reviewed (changes requested):
The code looks great.

let's use jsdoc for the comments though: https://jsdoc.app/

Use @property for all the properties on the config object: https://jsdoc.app/tags-property.html

and @example for the examples (which are great btw): https://jsdoc.app/tags-example.html

You also forgot the `innerText` property.

JohnAKASquib commented:
@mkcode line 64 is the innerText property isnt it?

JohnAKASquib committed:
added jsdoc for the comments

Lamarr20 commented on src/components/02-buttons/button.jsx:
```diff
@@ -1,60 +1,83 @@
-  ------------------------------
+  @example 
   const config = {
-    class: "usa-button--secondary",
```
`usa-button` is necessary when defining `const config` because in our button element the class attribute defines `usa-button` which is the default style sheet then all other `usa-button--variants` are defined after the default `class = "usa-button usa-button--secondary"` .
[uswds-button](https://designsystem.digital.gov/components/button/)

JohnAKASquib replied:
I see thanks for the link my bad Lamarr I'll fix it

JohnAKASquib commented on src/components/02-buttons/button.jsx:
```diff
@@ -0,0 +1,90 @@
+exports.render = function(config) {
+  return (
+    <button
+      class={`uswds-button${config.class}`}
```
@Lamarr20 The reason why I did that is because the class in the button always starts with "uswds-button"

JohnAKASquib committed:
Changed class property back

Changed class examples back to the way they were, and made default class usa-button not uswds-button

mkcode commented on src/components/02-buttons/button.jsx:
```diff
@@ -0,0 +1,90 @@
+    </button>
+  */
+}
+exports.render = function(config) {
```
```suggestion
exports.Button = function(config) {
```

mkcode committed:
Update src/components/02-buttons/button.jsx

mkcode commented on src/components/02-buttons/button.jsx:
```diff
@@ -0,0 +1,90 @@
+  /**  
+  ATTRIBUTES FOR THE BUTTON
+  ******************************
+  @namespace
```
@function
@parameter config

MathewDavidov replied:
```@param {*} config [description]```

mkcode replied:
```suggestion
@type { Function } Button
@param {Object} config [description]  
```

mkcode committed:
Update src/components/02-buttons/button.jsx

mkcode commented on src/components/02-buttons/button.jsx:
```diff
@@ -0,0 +1,91 @@
+  /**  
+  ATTRIBUTES FOR THE BUTTON
+  ******************************
+@type { Function } Button
+@param {Object} config [description]  
```
```suggestion
  @type { Function } Button
  @param {Object} config [description]  
```

mkcode committed:
Update src/components/02-buttons/button.jsx

mkcode reviewed (approved):
Great work. ✨
